pkg/storage/existing_model_linkStore.go

```diff
@@ -0,0 +1,96 @@
+package storage
+
+import (
+	"go.uber.org/zap"
+
+	"github.com/google/uuid"
+
+	"github.com/cmsgov/mint-app/pkg/models"
+	"github.com/cmsgov/mint-app/pkg/shared/utilitySQL"
+
+	_ "embed"
+)
+
+//go:embed SQL/existing_model_link/merge.sql
+var existingModelLinkMergeSQL string
+
+//go:embed SQL/existing_model_link/get_by_id.sql
+var existingModelLinkGetByIDSQL string
+
+//go:embed SQL/existing_model_link/get_by_model_plan_id_LOADER.sql
+var existingModelLinkGetByModelPlanIDLoaderSQL string
+
+// ExistingModelLinkGetByModelPlanIDLOADER returns the plan GeneralCharacteristics for a slice of model plan ids
+func (s *Store) ExistingModelLinkGetByModelPlanIDLOADER(logger *zap.Logger, paramTableJSON string) ([]*models.ExistingModelLink, error) {
+	linkSlice := []*models.ExistingModelLink{}
+
+	stmt, err := s.db.PrepareNamed(existingModelLinkGetByModelPlanIDLoaderSQL)
+	if err != nil {
+		return nil, err
+	}
+	arg := map[string]interface{}{
+		"paramTableJSON": paramTableJSON,
+	}
+
+	err = stmt.Select(&linkSlice, arg) //this returns more than one
+
+	if err != nil {
+		return nil, err
+	}
+
+	return linkSlice, nil
+}
+
+// ExistingModelLinksUpdate creates a new links that don't yet exist, deletes ones that are no longer provided,
+func (s *Store) ExistingModelLinksUpdate(logger *zap.Logger, userID uuid.UUID, modelPlanID uuid.UUID, existingModelIDs []int, currentModelPlanIDs []uuid.UUID) ([]*models.ExistingModelLink, error) {
+	tx := s.db.MustBegin()
+	defer tx.Rollback()
```

Comment:
```suggestion
	defer func(tx *sqlx.Tx) {
		var err = tx.Rollback()
		if err != nil {
			return nil, err
		}
	}(tx)
```

Reply:
interesting! This could be a good paradigm to explore, I think I'll leave out of this PR, but keep it in the back of my mind for next time we call the same functionality.